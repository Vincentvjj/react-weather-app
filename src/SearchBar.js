import React from 'react';
import style from './main.css';

const SearchBar = React.createClass({
    onClick() {
        this.props.getWeatherObj(this.refs.lat.value, this.refs.long.value);
    },

    render() {
        return(
            <div>
                Lat: <input type="text" placeholder="lat" ref="lat" className={style["margin-bottom"]}/>
                <br/>
                Long: <input type="text" placeholder="long" ref="long" className={style["margin-bottom"]}/>
                <br/>
                <button onClick={this.onClick}>Search</button>
            </div>
        );
    }

});

export default SearchBar;
