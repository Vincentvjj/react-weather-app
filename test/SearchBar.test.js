import React from 'react';
import style from '../src/main.css';
import SearchBar from '../src/SearchBar';
import renderer from 'react-test-renderer';

test('testing test', () => {


    const getWeatherObj = function () {
        return true;
    };
    const component = renderer.create(
        <SearchBar getWeatherObj={getWeatherObj}/>
    );

    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();

});