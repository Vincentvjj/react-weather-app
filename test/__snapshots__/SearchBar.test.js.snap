// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing test 1`] = `
<div>
  Lat: 
  <input
    className={undefined}
    placeholder="lat"
    type="text"
  />
  <br />
  Long: 
  <input
    className={undefined}
    placeholder="long"
    type="text"
  />
  <br />
  <button
    onClick={[Function]}
  >
    Search
  </button>
</div>
`;
